src/bundles/helia.js

```diff
@@ -0,0 +1,213 @@
+        apiOts: payload.apiOpts || state.apiOpts
+      })
+    }
+
+    if (type === 'IPFS_INIT_FAILED') {
+      return Object.assign({}, state, { ipfsReady: false, error: error })
+    }
+
+    return state
+  },
+
+  getExtraArgs () {
+    return { getIpfs: () => globalThis._ipfs }
+  },
+
+  selectIpfsReady: state => state.ipfs.ipfsReady,
+
+  selectIpfsIdentity: state => state.ipfs.identity,
+
+  doInitIpfs: () => async ({ dispatch, getState }) => {
+    console.log('Looking for IPFS')
+    console.time('IPFS_INIT')
+    dispatch({ type: 'IPFS_INIT_STARTED' })
+
+    let ipfs = null
+    let identity = null
+
+    // TODO: use delegatedContent&Peer routing via helia instead of kubo-rpc-client directly.
+    // TRY kubo-rpc-client (kubo-rpc-client)
+    const apiOpts = Object.assign(
+      {},
+      getState().ipfs.apiOpts,
+      getUserOpts('ipfsApi')
+    )
+    // TRY helia!
+    try {
+      console.time('HELIA_INIT')
+      console.log('Trying to start in-page helia')
+      console.info(
+        "🎛️ Customise your kubo-rpc-client opts by setting an `ipfsApi` value in localStorage. e.g. localStorage.setItem('ipfsApi', JSON.stringify({port: '1337'}))"
+      )
+      ipfs = await initHelia(apiOpts)
+      console.log('got ipfs')
+      // identity = await ipfs.libp2p.identify()
+      identity = 'helia-TBD'
+      console.log('in-page helia instance ready!', identity)
+      globalThis._ipfs = ipfs
+      console.timeEnd('HELIA_INIT')
+      console.timeEnd('IPFS_INIT')
+      return dispatch({
+        type: 'IPFS_INIT_FINISHED',
+        payload: {
+          identity,
+          provider: 'helia'
+        }
+      })
+    } catch (error) {
+      if (error.message && error.message.includes('subtle is undefined')) {
+        console.warn('IPLD Explorer requires access to window.crypto, redirecting to canonical URL that is known to provide it in all browsers')
+        // This error means js-ipfs was loaded in a context that is not marked
+        // as Secure Context by the browser vendor.  (example: *.localhost in
+        // Firefox until https://bugzilla.mozilla.org/show_bug.cgi?id=1220810
+        // is addresssed)
+        // This is difficult to debug for regular user, as Explorer simply fails to load anything from IPFS.
+        // For now, we detect this failure and redirect to canonical version with TLS, so it always works.
+        const url = new URL('https://explore.ipld.io/?x-ipfs-companion-no-redirect')
+        // url.hash = window.location.hash
+        // window.location.replace(url.toString())
+        return
+      }
+      console.log('Failed to initialise helia', error)
+      console.timeEnd('IPFS_INIT')
+      return dispatch({ type: 'IPFS_INIT_FAILED', error })
+    }
+  }
+}
+export default bundle
+
+async function initHelia (ipfsApi) {
+
+  const delegatedClient1 = kuboClient(ipfsApi)
+  const delegatedClient2 = kuboClient({
+    protocol: 'https',
+    port: 443,
+    host: 'node3.delegate.ipfs.io'
+  })
```

Comment:
@SgtPooki  yes, it is ok to use this RPC endpoint for the time being, but you should load-balance across all four nodes (maybe pick random one on page load?)

```
node3.delegate.ipfs.io
node2.delegate.ipfs.io
node1.delegate.ipfs.io
node0.delegate.ipfs.io
```


----

FYSA some long term plans: we want to switch away from Kubo PRC to IPNI/cid.contact, but that is hard to estimate, involves spec work, libraries, and services, incl. Kubo and cid.contact to hop onboard, and every team has different priotities. 

Realistically, first, we will add peer routing to  `/routing/v1/peers` spec (no IPIP yet, but soon), and have that exposed in Kubo, so `*.delegate.ipfs.io/routing/v1` is used instead of the Kubo RPC (`/api/v0`) and we use modern API with proper Cache-Control. Eventually, we will switch to `cid.contact/routing/v1/peers`, once they support it, but no ETA on that yet.

Reply:
I ended up removing this because it was causing more networking noise that we don't need. sticking to just the Kubo node and gateway; removing DHT, gossipsub, relay, bootstrapping, etc, and now we should only ever have 3 requests in flight. we can do it smarter later if we need to but this works well for now